client/components/data/query-order-transaction/index.jsx

```diff
@@ -0,0 +1,78 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { fetchOrderTransaction } from 'state/order-transactions/actions';
+import {
+	getOrderTransaction,
+	getOrderTransactionError,
+	isFetchingOrderTransaction,
+} from 'state/selectors';
+
+class QueryOrderTransaction extends React.Component {
+	static propTypes = {
+		orderId: PropTypes.number.isRequired,
+		pollIntervalMs: PropTypes.number,
+		transaction: PropTypes.object,
+		fetchTransaction: PropTypes.func.isRequired,
+		error: PropTypes.object,
+		isFetching: PropTypes.bool,
+	};
+
+	static defaultProps = {
+		pollIntervalMs: 0,
+	};
+
+	canFetch = () => {
+		const { error, isFetching } = this.props;
+
+		// fetch if no error has occurred and a fetch is not in progress.
+		return null == error && ! isFetching;
+	};
+
+	componentDidMount() {
+		const { pollIntervalMs, orderId, fetchTransaction } = this.props;
+
+		if ( pollIntervalMs ) {
+			this.timer = setInterval( () => {
+				if ( this.canFetch() ) {
+					fetchTransaction( orderId );
+				}
+			}, pollIntervalMs );
+			return;
+		}
+
+		if ( this.canFetch() ) {
+			fetchTransaction( orderId );
+		}
+	}
+
+	componentWillUnmount() {
+		if ( this.timer ) {
+			clearInterval( this.timer );
+		}
+	}
+
+	render() {
+		return null;
+	}
+}
+
+export default connect(
+	( state, props ) => ( {
+		transaction: getOrderTransaction( state, props.orderId ),
```

Comment:
This may be something about Query components that I don't fully grok, but what is the purpose of having the `transaction` included as a prop? Is there some side effect we need to occur?

Reply:
Argh! Thanks for the eagle eyes, @sirbrillig . I've removed it in 9a5ba59.